File: Week 02/id_081/BinaryTreeInorderTraversal.java
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

/**
 * BinaryTreeInorderTraversal
 */
public class BinaryTreeInorderTraversal {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> list = new ArrayList()<>();
        Stack<TreeNode> stack = new Stack()<>();
        TreeNode cur = root;
        while(cur!=null || !stack.isEmpty()){
            if(cur != null){
                stack.push(cur);
                cur = cur.left;
            }else{
                TreeNode node = stack.pop();
                list.add(node.val);
                cur = node.right;
            }
        }
        return list;
    }
}

File: Week 02/id_591/binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    let number = [];
    if(root == null){
        return number;
    }
        
    var inorder = function(binaryTree){
        if(binaryTree.left != null){
            inorder(binaryTree.left);
        }
        
        number.push(binaryTree.val);

        if(binaryTree.right != null){
            inorder(binaryTree.right);
        }
    }
    inorder(root);
    return number;
    
};
